diff --git a/Dockerfile b/Dockerfile
index b97a905..dbb6575 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.

-FROM golang:1.20.6@sha256:cfc9d1b07b1ef4f7a4571f0b60a99646a92ef76adb7d9943f4cb7b606c6554e2 AS builder
+FROM registry.access.redhat.com/ubi9/ubi-micro AS builder
 ENV APP_ROOT=/opt/app-root
 ENV GOPATH=$APP_ROOT

